pkg/clusters/addons/kong/builder.go

```diff
@@ -36,6 +40,28 @@ func (b *Builder) WithPostgreSQL() *Builder {
 	return b
 }
 
+// WithEnterprise deploying kong enterpise
+func (b *Builder) WithEnterprise() *Builder {
+	b.enterprise = true
+	b.repo = DefaultEnterpriseImageRepo
+	b.tag = DefaultEnterpriseImageTag
+	b.license = KongLicenseSecretName
+	return b
+}
+
+// WithImage specify docker image repo and tag
+func (b *Builder) WithImage(repo, tag string) *Builder {
+	b.repo = repo
+	b.tag = tag
+	return b
+}
+
+// WithLicense specify licese secret name
+func (b *Builder) WithLicense(license string) *Builder {
+	b.license = license
+	return b
+}
```

Comment:
I guess I'm not opposed to having the license SETTER method be separate from the Enterprise setter, but for my own edification can you help me to understand why these are separate? It seemed to me that you could just as easily do `WithEnterprise(license string)` and this would take care of all of it in one shot? :thinking:

Reply:
if user already installed a license in the cluster, just pass the name; otherwise, we created default license for test purpose.

Reply:
:thinking: OK, I guess it does help to support an end-user who wants to install the license their own way, but provides a standard way otherwise. I'm persuaded :+1: